tests/ui/implicit_return.stderr

```diff
@@ -1,64 +1,76 @@
 error: missing return statement
-  --> $DIR/implicit_return.rs:8:5
+  --> $DIR/implicit_return.rs:9:5
    |
 LL |     true
    |     ^^^^ help: add `return` as shown: `return true`
    |
    = note: `-D clippy::implicit-return` implied by `-D warnings`
 
 error: missing return statement
-  --> $DIR/implicit_return.rs:14:9
+  --> $DIR/implicit_return.rs:15:9
    |
 LL |         true
    |         ^^^^ help: add `return` as shown: `return true`
 
 error: missing return statement
-  --> $DIR/implicit_return.rs:16:9
+  --> $DIR/implicit_return.rs:17:9
    |
 LL |         false
    |         ^^^^^ help: add `return` as shown: `return false`
 
 error: missing return statement
-  --> $DIR/implicit_return.rs:24:17
+  --> $DIR/implicit_return.rs:25:17
    |
 LL |         true => false,
    |                 ^^^^^ help: add `return` as shown: `return false`
 
 error: missing return statement
-  --> $DIR/implicit_return.rs:25:20
+  --> $DIR/implicit_return.rs:26:20
    |
 LL |         false => { true },
    |                    ^^^^ help: add `return` as shown: `return true`
 
 error: missing return statement
-  --> $DIR/implicit_return.rs:40:9
+  --> $DIR/implicit_return.rs:41:9
    |
 LL |         break true;
    |         ^^^^^^^^^^ help: change `break` to `return` as shown: `return true`
 
 error: missing return statement
-  --> $DIR/implicit_return.rs:48:13
+  --> $DIR/implicit_return.rs:49:13
    |
 LL |             break true;
    |             ^^^^^^^^^^ help: change `break` to `return` as shown: `return true`
 
 error: missing return statement
-  --> $DIR/implicit_return.rs:57:13
+  --> $DIR/implicit_return.rs:58:13
    |
 LL |             break true;
    |             ^^^^^^^^^^ help: change `break` to `return` as shown: `return true`
 
 error: missing return statement
-  --> $DIR/implicit_return.rs:75:18
+  --> $DIR/implicit_return.rs:76:18
    |
 LL |     let _ = || { true };
    |                  ^^^^ help: add `return` as shown: `return true`
 
 error: missing return statement
-  --> $DIR/implicit_return.rs:76:16
+  --> $DIR/implicit_return.rs:77:16
    |
 LL |     let _ = || true;
    |                ^^^^ help: add `return` as shown: `return true`
 
-error: aborting due to 10 previous errors
+error: missing return statement
+  --> $DIR/implicit_return.rs:81:5
+   |
+LL |     panic!()
```

Comment:
Should we really suggest `return panic!()`? I think this is misleading, since it suggests that you can return something you get from the `panic!()` call. But `panic!()` never returns, since it kills the program. This lint is mainly for "Programmers coming from other programming languages" (= not familiar with the language), so there is even a higher probability that this gets misinterpreted.

Reply:
> Should we really suggest `return panic!()`? I think this is misleading, since it suggests that you can return something you get from the `panic!()` call. But `panic!()` never returns, since it kills the program.

With this PR the following will NOT suggest `return`:
```rust
fn blubb() -> bool {
  panic!()
}
```
but this WILL suggest `return`:
```rust
fn blubb() -> ! {
  panic!()
}
```
Before this PR both examples would suggest `return`.
> What are your thoughts about this?

I do agree that suggesting `return` on panic doesn't make sense, thats mainly what got fixed in this PR. I'm handling returning the never type (`!`) as a special case where it does make sense to me to suggest `return`.
`!` does actually not represent panics, but just that the function *never* returns.
This is really opinionated I believe, so really any further input is welcome, personally this is how I like it.

> This lint is mainly for "Programmers coming from other programming languages" (= not familiar with the language)

I'm still using it because I dislike the lack of verbosity with only skipping the `;` to signify return, but I would bet that I'm one of very very few people who use this lint 😄.

Reply:
> I'm handling returning the never type (`!`) as a special case where it does make sense to me to suggest return.
>`!` does actually not represent panics, but just that the function _never_ returns.

That's exactly my point. `!` means that this function _never_ returns, so why suggest to add a `return`? I personally think that's confusing 😄 

> Before this PR both examples would suggest return.

That's why I would just not lint at all on expressions like
```rust
fn nr() -> ! {
    // some code ...
    fn_that_never_returns()
}
```

Reply:
I removed with the latest [commit](https://github.com/rust-lang/rust-clippy/pull/4274/commits/a2b63af74653ec4d7324f55aa8edf2460e739e1f).

Also noticed that I mispelled "linting" in the commit 😄.